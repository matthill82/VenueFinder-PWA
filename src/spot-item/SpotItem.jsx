import React from "react";
import PropTypes from "prop-types";

const SpotItem = props => (
  <li key={`item-${props.id}`} data-selector="spot-table__item" className="spot-table__item">
    <button className="spot-table__item-name">{props.name}</button>
    <button className="spot-table__item-distance">
      {(props.distance / 1000).toFixed(2)}km away
    </button>
  </li>
);

SpotItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  distance: PropTypes.number.isRequired
};

export default SpotItem;
