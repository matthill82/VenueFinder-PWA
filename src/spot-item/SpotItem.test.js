import React from "react";
import SpotItem from "./SpotItem";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("The SpotItem component", () => {
  const props = {
    id: "d29348f",
    name: "test",
    address: "address",
    distance: 1234
  };

  it("renders correctly", () => {
    const tree = renderer.create(<SpotItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const component = shallow(<SpotItem {...props} />);
    expect(component.exists()).toEqual(true);
  });
});
