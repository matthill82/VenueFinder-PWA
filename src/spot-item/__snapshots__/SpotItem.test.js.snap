// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SpotItem component renders correctly 1`] = `
<li
  className="spot-table__item"
  data-selector="spot-table__item"
>
  <button
    className="spot-table__item-name"
  >
    test
  </button>
  <button
    className="spot-table__item-distance"
  >
    1.23
    km away
  </button>
</li>
`;
